lvsfunc/util.py

```diff
@@ -174,19 +214,20 @@ def replace_ranges(clip_a: vs.VideoNode,
     if ranges is None:
         return clip_a
 
-    out = clip_a
+    out_indicies = list(zip([0] * clip_a.num_frames, range(clip_a.num_frames)))
+    clip_b_indices = list(zip([1] * clip_b.num_frames, range(clip_b.num_frames)))
 
     nranges = normalize_ranges(clip_b, ranges)
 
     for start, end in nranges:
```

Comment:
since we're now using a normal `List` instead of a `VideoNode`, we can just do slice assignment
```python
for start, end in nranges:
    out_indices[start:end+1] = list(zip([1] * (end + 1 - start), range(start, end+1))
```

Reply:
Yup you've lost me now, please post the full block that this is relevant to so I can better understand.

Reply:
```python
out_indicies = [(0, i) for i in range(clip_a.num_frames)]

for start, end in nranges:
    out_indices[start:end+1] = [(1, i) for i in range(start, end+1)]
```

can replace the entire for loop